### DISCUSSION TITLE
The following error occurred in the Problem 'MOOSE Problem' of type FEProblem.

### DISCUSSION THREAD

**Discussion by engrmessi** [idaholab/moose#27266] [Q&A Modules: Solid mechanics]
my scripts was running very well untill i updated my moose. I encounter this challange. Can anyone help me fix it

*** ERROR ***
The following error occurred in the Problem 'MOOSE Problem' of type FEProblem.
Material property 'acc_slip', requested by 'accslip' is not defined on block 0
Material property 'lage', requested by 'e_xx' is not defined on block 0
Material property 'fp', requested by 'fp_xx' is not defined on block 0
Material property 'gss', requested by 'gss1' is not defined on block 0
[39m

[31m
*** ERROR ***
The following error occurred in the Problem 'MOOSE Problem' of type FEProblem.

Material property 'acc_slip', requested by 'accslip' is not defined on block 0
Material property 'lage', requested by 'e_xx' is not defined on block 0
Material property 'fp', requested by 'fp_xx' is not defined on block 0
Material property 'gss', requested by 'gss1' is not defined on block 0
[39m

** here is my script which was working before the update
[GlobalParams]
  displacements = 'disp_x disp_y'
[]

[Mesh]
  type = GeneratedMesh
  dim = 2
  nx = 40
  ny = 20
  xmin = 0
  xmax = 20
  ymin = 0
  ymax = 10
[]

[Variables]
  [./eta1]
    order = FIRST
    family = LAGRANGE
  [../]
  [./disp_x]
    block = 0
  [../]
  [./disp_y]
    block = 0
  [../]
[]

[AuxVariables]
  [./bounds_dummy]
    order = FIRST
    family = LAGRANGE
  [../]
  [./accslip]
    order = CONSTANT
    family = MONOMIAL
    block = 0
  [../]
  [./vonmises]
    order = CONSTANT
    family = MONOMIAL
    block = 0
  [../]
  [./stress_xx]
    order = CONSTANT
    family = MONOMIAL
    block = 0
  [../]
  [./fp_xx]
    order = CONSTANT
    family = MONOMIAL
    block = 0
  [../]
  [./e_xx]
    order = CONSTANT
    family = MONOMIAL
    block = 0
  [../]
  [./gss1]
    order = CONSTANT
    family = MONOMIAL
    block = 0
  [../]
[]

[Bounds]
  [./eta_upper_bound]
    type = ConstantBoundsAux
    variable = bounds_dummy
    bounded_variable = eta1
    bound_type = upper
    bound_value = 1
  [../]
  [./eta_lower_bound]
    type = ConstantBoundsAux
    variable = bounds_dummy
    bounded_variable = eta1
    bound_type = lower
    bound_value = 0
  [../]
[]

[ICs]
  [./eta1]
    variable = eta1
    type = SmoothCircleIC
    x1 = 10
    y1 = 10
    int_width = 0.05
    radius = 0.5
    invalue = 0
    outvalue = 1
    #int_width = 1
  [../]
[]

[Functions]
  [./tdisp]
    type = ParsedFunction
    expression = 0.005*t
  [../]
[]

[Modules/TensorMechanics/Master/all]
  strain = FINITE
  add_variables = true
  planar_formulation = PLANE_STRAIN        
#  generate_output = stress_zz
  strain_base_name = uncracked
[]

[Kernels]
  # Allen-Cahn Equation for eta1
  [./detadt_eta1]
    type = TimeDerivative
    variable = eta1
  [../]
  [./ACinterface_eta1]
    type = ACInterface
    variable = eta1
    mob_name = L_inf
    kappa_name = kappa
  [../]
  [./ACBulkF_eta1]
    type = AllenCahn
    variable  = eta1
    f_name = g1
    mob_name = L_inf
#    args = T
  [../]
  [./AllenCahn1]
    type = AllenCahn
    variable = eta1
    mob_name = L_pot
    f_name = f1
#    args = T
  [../]

[]

[AuxKernels]
  [./accslip]
    type = MaterialRealAux
    variable = accslip
    property = acc_slip
    execute_on = timestep_end
    block = 0
  [../]
  [./vonmises]
    type = RankTwoScalarAux
    rank_two_tensor = stress
    variable = vonmises
    scalar_type = VonMisesStress
    execute_on = timestep_end
    block = 0
  [../]
  [./stress_xx]
    type = RankTwoAux
    rank_two_tensor = stress
    variable = stress_xx
    index_j = 0
    index_i = 0
    execute_on = timestep_end
    block = 0
  [../]
  [./fp_xx]
    type = RankTwoAux
    variable = fp_xx
    rank_two_tensor = fp
    index_j = 0
    index_i = 0
    execute_on = timestep_end
    block = 0
  [../]
  [./e_xx]
    type = RankTwoAux
    variable = e_xx
    rank_two_tensor = lage
    index_j = 0
    index_i = 0
    execute_on = timestep_end
    block = 0
  [../]
  [./gss1]
    type = MaterialStdVectorAux
    variable = gss1
    property = gss
    index = 0
    execute_on = timestep_end
    block = 0
  [../]
[]

[BCs]
  [./eta_bottom]
    type =  DirichletBC
    variable = 'eta1'
    boundary = 'bottom'
    value = 1.0
  [../]
  [./eta_lr]
    type =  NeumannBC
    variable = 'eta1'
    boundary = 'top left right'
    value = 0.0
  [../]
  [./Pressure]
    [./stressfree_boundary]
       #Applies the pressure
       boundary = top
       factor = 0.0 
#       disp_x = disp_x
#       disp_y = disp_y
    [../]
  [../]
  [./dispy]
    type = DirichletBC
    variable = disp_y
    boundary = bottom
    value = 0
  [../]
  [./dispx]
    type = DirichletBC
    variable = disp_x
    boundary = left
    value = 0
  [../]
  [./tdisp]
    type = FunctionDirichletBC
    variable = disp_x
    boundary = right
    function = tdisp
  [../]
[]

[Materials]
  [./crysp]
    type = FiniteStrainCPSlipRateRes
    block = 0
    slip_sys_file_name = input_slip_sys.txt
    nss = 12
    num_slip_sys_flowrate_props = 2 #Number of properties in a slip system
    flowprops = '1 4 0.5 0.05 5 8 0.5 0.05 9 12 0.5 0.05'
    hprops = '1.0 215 78.6 200.0 1.75'
    gprops = '1 4 78.6 5 8 78.6 9 12 78.6'
#    tan_mod_type = exact
    slip_incr_tol = 1
    maximum_substep_iteration = 12
#    use_line_search = true
#    line_search_method = 'BISECTION'
    rtol = 1e-7
    abs_tol = 1e-8
    base_name = uncracked
  [../]
  [./elasticity_tensor]
    type = ComputeElasticityTensorCP
    block = 0
    C_ijkl = '2.04e5 1.37e5 1.37e5 2.044e5 1.37e5 2.04e5 1.26e5 1.26e5 1.26e5'
    fill_method = symmetric9
#    read_prop_user_object = prop_read
    euler_angle_1 = 0
    euler_angle_2 = 0
    euler_angle_3 = 0
    output_properties = 'Euler_angles'
    outputs = exodus
    base_name = uncracked
  [../]
  [./cracked_stress]
    type = ComputeCrackedStressmod
    c = eta1
#    property_name = E_el
    use_current_history_variable = false
    uncracked_base_name = uncracked
    finite_strain_model = false
  [../]
  [./degradation]
    type = DerivativeParsedMaterial
    property_name = degradation
    coupled_variables = 'eta1'
#    function = '1'
    expression = 'eta1^3*(6*eta1^2-15*eta1+10)*(1.0 - c) + c'
    constant_names       = 'c'
    constant_expressions = '1e-6'
    derivative_order = 2
  [../]
  [./f1]
    type = DerivativeParsedMaterial
    property_name = f1
    coupled_variables = 'eta1'
#    material_property_names = 'h0'
#                diff_factor:=if(c>c_b,exp(alpha*log(c/c_b)),1);
    constant_names = 'overpotential  R     z     alpha beta F_const T '
    constant_expressions = '0.1    8.31    2.19  0.65  0.35  96485.33 298'
    expression = 'h1:=eta1^3*(6*eta1^2-15*eta1+10);
    h1*(exp(z*alpha*F_const*overpotential/(R*T))-exp(-z*beta*F_const*overpotential/(R*T)))'
#    outputs = exodus
    derivative_order = 1
  [../]
  [./h1]
     type = SwitchingFunctionMaterial
     h_order = HIGH
     eta = 'eta1'
     function_name = h1
  [../]
  [./g1]
     type = BarrierFunctionMaterial
     g_order = SIMPLE
     eta = 'eta1'
     function_name = g1
  [../]
  [./constants]
    type = GenericConstantMaterial
    prop_names  = 'kappa     L_inf0    L_pot0      W'
    prop_values = '0.05      0.0009   0.00415      1.0'  #0.99e-4  4.15e-4
  [../]
  [./reaction_mob]
    type = ParsedMaterial
    property_name = L_inf
    coupled_variables = 'accslip'
    material_property_names = 'L_inf0'
#    function = 'L_inf0'
    expression = 'L_inf0*(1+accslip/0.001)'
  [../]
  [./reaction_mob2]
    type = ParsedMaterial
    property_name = L_pot
    coupled_variables = 'accslip'
    material_property_names = 'L_pot0'
#    function = 'L_pot0'
    expression = 'L_pot0*(1+accslip/0.001)'
  [../]
[]

[Postprocessors]
  [./vonmises]
    type = ElementAverageValue
    variable = vonmises
    block = 'ANY_BLOCK_ID 0'
  [../]
  [./stress_xx]
    type = ElementAverageValue
    variable = stress_xx
    block = 'ANY_BLOCK_ID 0'
  [../]
  [./fp_xx]
    type = ElementAverageValue
    variable = fp_xx
    block = 'ANY_BLOCK_ID 0'
  [../]
  [./e_xx]
    type = ElementAverageValue
    variable = e_xx
    block = 'ANY_BLOCK_ID 0'
  [../]
  [./gss1]
    type = ElementAverageValue
    variable = gss1
    block = 'ANY_BLOCK_ID 0'
  [../]
[]

[Preconditioning]
  [./smp]
    type = SMP
    full = true
  [../]
[]

[Executioner]
  type = Transient

  #Preconditioned JFNK (default)
  solve_type = 'PJFNK'

  petsc_options_iname = '-pc_type -pc_factor_mat_solver_package -snes_type'
  petsc_options_value = 'lu            superlu_dist            vinewtonrsls'
  automatic_scaling = true
  
  l_max_its = 20
  nl_max_its = 10
  nl_abs_tol = 1e-9

#  petsc_options_iname = '-pc_type -pc_asm_overlap -sub_pc_type -ksp_type -ksp_gmres_restart'
#  petsc_options_value = ' asm      1              lu            gmres     200'

  dt = 0.0025
  dtmax = 0.01

  [./TimeStepper]
    type = IterationAdaptiveDT
    dt = 0.0025 # Initial time step.  In this simulation it changes.
    optimal_iterations = 8 # Time step will adapt to maintain this number of nonlinear# iterations
  [../]
  [./Adaptivity]
    initial_adaptivity = 2 # Number of times mesh is adapted to initial condition
    refine_fraction = 0.6 # Fraction of high error that will be refined
    coarsen_fraction = 0.2 # Fraction of low error that will coarsened
    max_h_level = 2 # Max number of refinements used, starting from initial mesh (befo# uniform refinement)
  [../]

  end_time = 5

[]

[VectorPostprocessors]
  [eta1]
    type = LineValueSampler
    start_point = '10 0 0'
    end_point = '10 10 0'
    num_points = 200
    sort_by = y
    variable = eta1
  []
[]

[Outputs]
  exodus = true
  interval = 2
  [./console]
     type = Console
     interval = 2
  [../]
  [./csv]
     type = CSV
     interval = 2
  [../]
[]

**Comment by GiudGiud** [comment]
Hello

please use Debug/show_material_props to see a summary of producers and consumers of material properties and make sure they match

Guillaume

**Comment by engrmessi** [comment]
sorry, i did not understand what you mean. where should implement this " Debug/show_material_props"? in my input script or bash file?

Best!

**Reply by GiudGiud** [reply]
The Debug block is a block you can add to the input file

see the SetupDebugAction documentation for more info on the parameters
